chain/network/src/peer_manager.rs

```diff
@@ -376,8 +396,11 @@ impl PeerManagerActor {
     }
 
     /// Get a random peer we are not connected to from the known list.
-    fn sample_random_peer(&self, ignore_list: &HashSet<PeerId>) -> Option<PeerInfo> {
-        let unconnected_peers = self.peer_store.unconnected_peers(ignore_list);
+    fn sample_random_peer(&self) -> Option<PeerInfo> {
+        let unconnected_peers = self.peer_store.unconnected_peers(|peer_id| {
+            // Ignore peers in active set and in outgoing list
+            self.active_peers.contains_key(&peer_id) || self.outgoing_peers.contains(&peer_id)
+        });
         let index = thread_rng().gen_range(0, std::cmp::max(unconnected_peers.len(), 1));
```

Comment:
This feels unnecessary. You can directly sample an element from the array using `choose`.